WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -47,7 +47,9 @@ def index
     end
 
     if params[:years] == "current"
-      @competitions = @competitions.where("CAST(CONCAT(year,'-',month,'-',day) as Datetime) > ?", (Date.today - PAST_COMPETITIONS_DAYS))
+      @competitions = @competitions.select do |competition|
```

Comment:
While I do like how much more readable this is, this will require Ruby walking over all ~2k competitions, rather than mysql, where we might be able to benefit from an index. Granted, there probably isn't an index on dates right now, but that's something I see us doing later after we clean up the database schema.